cfme/test_framework/sprout/client.py

```diff
@@ -111,7 +111,8 @@ def provision_appliances(
             group=stream, provider=provider, lease_time=lease_time, ram=ram, cpu=cpu, count=count
         )
         wait_for(
-            lambda: self.call_method('request_check', str(request_id))['finished'], num_sec=300)
+            lambda: self.call_method('request_check', str(request_id))['finished'], num_sec=300,
```

Comment:
if we remove the `['finished']`
and use `fail_condition=lambda result: not result['finished']` instead,
the next call is no longer needed and it can be spelled

```
data = wait_for(
       lambda: self.call_method('request_check', str(request_id)), 
       num_sec=300, fail_condition=lambda result: not result['finished'],
       message='provision {} appliance(s) from sprout'.format(count)).out
````